content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx

```diff
@@ -0,0 +1,186 @@
+}
+```
+
+You can also `@map` an enum value, or `@@map` an enum:
+
+```prisma highlight=3,5;normal
+enum Type {
+  Blog,
+  Twitter @map("comment_twitter")
+
+  @@map("comment_source_enum")
+}
+```
+
+`@map` and `@@map` allow you to [tune the shape of your Prisma Client API](../prisma-client/working-with-prismaclient/use-custom-model-and-field-names#using-map-and-map-to-rename-fields-and-models-in-the-prisma-client-api) by decoupling model and field names from table and column names in the underlying database.
+
+## Constraint and index names (Preview)
+
+In [2.29.0](https://github.com/prisma/prisma/releases/tag/2.29.0) and later, you can use the `map` field to define the **underlying constraint and index names** in the Prisma schema. The `map` field is _only_ visible in the schema if you use a constraint name that differs from Prisma's naming convention.
+
+<Admonition type="warning">
+
+**Not supported by MongoDB**<br />
+Custom constraint and index names apply to relational databases only.
+
+</Admonition>
+
+### Enabling the <inlinecode>namedConstraints</inlinecode> Preview feature
+
+Named constraints is a Preview feature. To enable this feature:
+
+1. Add `namedConstraints` to the `previewFeatures` field in the `generator` block:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["namedConstraints"]
+}
+```
+
+2. Choose an [upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2290#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
+
+<Admonition type="alert">
+
+If you use Prisma Migrate and want to maintain your existing constraint and index names, **do not** enable the `namedConstraints` feature and immediately run `prisma migrate dev`. This will **change any underlying constraint name that does not follow Prisma's convention**. Follow the [upgrade path for the Prisma Migrate workflow](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2290#upgrade-paths-for-prisma-migrate-users).
+
+</Admonition>
+
+### Use cases for named constraints
+
+Some use cases for explicitly named constraints include:
+
+- Company policy
+- Conventions of other tools
+
+### Using custom constraint names
+
+The following schema defines three constraints (`@id`, `@unique`, and `@relation`) and one index (`@@index`):
+
+```prisma highlight=2,8,11,14;normal
+model User {
+  id    Int    @id @default(autoincrement())
+  name  String @unique
+  posts Post[]
+}
+
+model Post {
+  id          Int     @id @default(autoincrement())
+  title       String
+  authorName  String  @default("Anonymous")
+  author      User?   @relation(fields: [authorName], references: [name])
+}
+
+@@index([title, authorName])
+```
+
+By default, constraint and index names in the underlying database follow Prisma's naming convention. Constraint or index names that follow Prisma's naming convention are not included in the schema.
```

Comment:
Information about the defaults listed here? Would be the perfect location @do4gr - maybe a list or table.

Reply:
Also:
```suggestion
By default, constraint and index names in the underlying database follow Prisma's naming convention.

Constraint or index names that follow Prisma's naming convention are not included in the schema.
```
The second sentence is weird here as it mostly refers to Introspection (see a change I suggested above for `@(@)map`), so maybe again mention that?

Reply:
I'd even say that sentence is could be understood wrong. Dbs created by migrate follow prismas naming convention when no explicit names are specified. Introspection just reads from the db and then does not render them if they follow our naming convention.

Reply:
Spelled out our convention and rephrased the sentence.